### PR TITLE
Dynamic Types Configuration [15079]

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

File: docs/rst/exports/alias.include [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .. |FastDDSDocs| replace:: `Fast DDS Documentation <https://fast-dds.docs.eprosima.com/en/v2.3.0/>`__
 .. |DDSSpecification| replace:: `DDS specification <https://www.omg.org/spec/DDS/1.4/About-DDS/>`__
+.. |ddsvisualizer| replace:: *DDS Visualizer Plugin*
+.. |fastdds| replace:: *Fast DDS*
```
Comment (antonrv) [inline docs/rst/exports/alias.include:4]: add also a link to FastDDS documentation?

File: docs/rst/user_manual/dynamic_data_types.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+If the application does not know the data type, it will not be able to connect to such topic.
+There are two ways of retrieving the data type information to the application:
+
+********************************
+Automatically discover Data Type
+********************************
+
+In order for |ddsvisualizer| to discover automatically the data type of a topic,
+the |fastdds| publisher that may interact with the application must be configured to send this information.
+There are two main configurations that has to be taken into account:
```
Comment (antonrv) [inline docs/rst/user_manual/dynamic_data_types.rst]: * that **have**
* I'd say that have **to be considered**

File: docs/rst/user_manual/dynamic_data_types.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+
+Generate DataType
+=================
+
+In order for Fast DDS to be able to retrieve the data type description, it must have the TypeInformation.
```
Comment (antonrv) [inline docs/rst/user_manual/dynamic_data_types.rst]: alias `|fastdds|` for consistency

File: docs/rst/user_manual/dynamic_data_types.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+Even when it is configured by default, make sure the :code:`auto_fill_type_information` value of the
+:code:`TypeSupport` is :code:`true`.
+
+
+*************
+XML Data Type
+*************
+
+If the automatic discovery of the data type is not possible, an XML file with the data type description could be loaded,
+so the internal |ddsvisualizer| is able to know the data type, and so, to subscribe to the topic.
```
Comment (antonrv) [inline docs/rst/user_manual/dynamic_data_types.rst]: I'd say **is able to know the data type to subscribe to the topic**

File: docs/rst/user_manual/dynamic_data_types.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+
+.. warning::
+
+    XML Data Type Description does not support keyed topics.
+
+
+.. warning::
+
+    Dynamic Types in Fast DDS does not support complex data structures of kind: :code:`sequence<struct>`.
```
Comment (antonrv) [inline docs/rst/user_manual/dynamic_data_types.rst]: **does not** -> **do not**

File: docs/rst/user_manual/dynamic_data_types.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+In order for Fast DDS to be able to retrieve the data type description, it must have the TypeInformation.
+In case that the |fastdds| application is already using the *Dynamic Types* API, this TypeInformation will be
+already available for the :code:`DomainParticipant`.
+
+Otherwise, if the data type is generated via
+`Fast DDS Gen <https://fast-dds.docs.eprosima.com/en/latest/fastddsgen/introduction/introduction.html>`_,
+it is required to generate it with option :code:`-typeobject`.
+This generates two auxiliar cpp files that include the type information of the data type.
+If doing so, the data must be registered in the |fastdds| application before creating the publisher,
```
Comment (antonrv) [inline docs/rst/user_manual/dynamic_data_types.rst]: **If doing so,** the data...  -> sounds strange, I'd use any of the alternatives:
* **Then,** the data...
* **After that,** the data...
* **When this is done,** the data...

**Review by antonrv — COMMENTED** [review]
Some styling comments and minor typos

File: docs/rst/exports/alias.include (hunk 1 above)
Comment (jparisu) [reply docs/rst/exports/alias.include:4]: It exist already, first line of the document

**Review by antonrv — APPROVED** [review]
good to me

**Pull request merged**